languages/wikipron/src/write.py

```diff
@@ -0,0 +1,97 @@
+import os
+import json
+
+LANGUAGES_PATH = "languages.json"
+
+
+def _readme_insert(wiki_name, row_string):
+    with open("../tsv/README.md", "r+") as source:
+        readme_list = []
+        readme_text = source.read()
+        readme_list.extend(readme_text.splitlines(True))
+        length = len(readme_list)
+        wiki_name_of_last_entry = readme_list[-1].split("|")[4][1:-1]
+        # If the table is only a newline, a row of headers
+        # and a formatting row or if the Wiktionary name
+        # of the last entry is alphabetically prior to
+        # the name we are adding.
+        if (
+            length == 3
+            or wiki_name > wiki_name_of_last_entry
+        ):
+            # Write the input string to the end
+            source.write(row_string)
+        else:
+            for i in range(3, length):
+                isolated_name = readme_list[i].split("|")[4][1:-1]
+                # Replaces old row.
+                if (
+                    isolated_name == wiki_name
+                    and row_string.split("|")[6][1:-1] in readme_list[i]
+                ):
+                    readme_list[i] = row_string
+                    break
+                # Inserts new row.
+                if isolated_name > wiki_name:
+                    readme_list.insert(i, row_string)
+                    break
+            # Rewrite the entire README.
+            source.seek(0)
+            source.truncate()
+            source.write("".join(readme_list))
```

Comment:
If each element in `readme_list` was a line w/o newlines you could use the more-efficient `writelines` (which takes an iterator of strings and doesn't build up the whole big string at once). But this part is so complex I'm afraid to ask you to kick the hornet's nest...

Reply:
I'll look into this.